### PR TITLE
Improve error message when mpmath is not installed

### PR DESCRIPTION
**Pull request by asmeurer**
Another idea that I hadn't considered would be to include `sympy.mpmath` as a backwards compatible import (that maps to the external mpmath). I think a lot of code uses sympy.mpmath and it would ease transition. I'm not sure how to do it, though.

### CODE REVIEW

File: sympy/__init__.py [hunk 1]
```diff
@@ -10,6 +10,15 @@
 """
 
 from __future__ import absolute_import, print_function
+del absolute_import, print_function
+
+try:
+    import mpmath
+except ImportError:
+    raise ImportError("SymPy now depends on mpmath as an external library. "
+    "You can install mpmath with 'pip install mpmath'.")
```
Comment (moorepants) [inline sympy/__init__.py]: How about: "You can install mpmath with 'pip install mpmath' or see the SymPy documentation for other options."

Comment (asmeurer) [inline sympy/__init__.py]: Maybe "see http://docs.sympy.org/latest/install.html for more information" (except that url won't be correct until we release).

Comment (moorepants) [inline sympy/__init__.py]: Sure that's fine too.

**Comment by certik** [comment]
Yep, that was my original proposal as well, but I haven't figured out a way to do that. If you find a way, then we should definitely do it.

**Comment by asmeurer** [comment]
I think this does it, but I need to play around with it some more.

```diff
diff --git a/sympy/__init__.py b/sympy/__init__.py
index bc1a34a..a8d1faf 100644
--- a/sympy/__init__.py
+++ b/sympy/__init__.py
@@ -18,8 +18,6 @@
     raise ImportError("SymPy now depends on mpmath as an external library. "
     "You can install mpmath with 'pip install mpmath'.")

-del mpmath
-
 from sympy.release import __version__

 import sys
@@ -29,6 +27,8 @@
     pass
     # Here we can also check for specific Python 3 versions, if needed

+sys.modules['sympy.mpmath'] = mpmath
+
 del sys

```

File: sympy/__init__.py [hunk 2]
```diff
@@ -20,6 +27,8 @@
     pass
     # Here we can also check for specific Python 3 versions, if needed
 
+sys.modules['sympy.mpmath'] = mpmath
```
Comment (certik) [inline sympy/__init__.py]: I would not put this here. I would tell users that they can put this into their own code if needed, but sympy itself shouldn't have it.

File: sympy/__init__.py (hunk 1 above)
Comment (certik) [inline sympy/__init__.py]: Giving a link is fine.

File: doc/src/install.rst [hunk 3]
```diff
@@ -3,89 +3,58 @@
...
-    $ python setup.py install
-
-Alternatively, if you don't want to install the package onto your computer, you
-may run SymPy with the "isympy" console (which automatically imports SymPy
-packages and defines common symbols) by executing within the "sympy" folder::
+Anaconda
+========
 
-    $ ./bin/isympy
+`Anaconda <http://continuum.io/downloads>`_ is a free Python a free Python
```
Comment (certik) [inline doc/src/install.rst]: a free Python is there twice.

**Comment by certik** [comment]
+1 to merge as it is now.

**Pull request merged**